fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryAbstractDataModificationPlan.java

```diff
@@ -201,17 +206,17 @@ public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull final
                                                                      @Nonnull final ExecuteProperties executeProperties) {
         final RecordCursor<QueryResult> results =
                 getInnerPlan().executePlan(store, context, continuation, executeProperties.clearSkipAndLimit());
-        final var targetDescriptor = store.getRecordMetaData().getRecordType(targetRecordType).getDescriptor();
+        final var targetDescriptor = getTargetDescriptor(store);
         return results
                 .map(queryResult -> Pair.of(queryResult, mutateRecord(store, context, queryResult, targetDescriptor)))
-                .mapPipelined(pair -> saveRecordAsync(store, pair.getRight(), executeProperties.isDryRun())
-                                .thenApply(storedRecord -> {
+                .mapPipelined(pair -> saveRecordAsync(store, context, pair.getRight(), executeProperties.isDryRun())
+                                .thenApply(queryResult -> {
                                     final var nestedContext = context.childBuilder()
                                             .setBinding(inner.getAlias(), pair.getLeft()) // pre-mutation
-                                            .setBinding(currentModifiedRecordAlias, storedRecord.getRecord()) // post-mutation
+                                            .setBinding(currentModifiedRecordAlias, queryResult.getMessage()) // post-mutation
                                             .build(context.getTypeRepository());
                                     final var result = computationValue.eval(store, nestedContext);
-                                    return QueryResult.ofComputed(result, null, storedRecord.getPrimaryKey(), null);
```

Comment:
Yeah, I fixed both `Update` and `Insert` plan such that they created a `QueryResult` from a `FDBQueriedRecord`, which is created from the return `FDBStoredRecord`.

This means, that both `Update` and `Insert` are now correctly announcing that they are returning `true` in their corresponding hooks in `StoredRecordProperty`.

The `DeletePlan` however is special because it is returning whatever the underlying plan is returning:

```java
 public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull final FDBRecordStoreBase<M> store,
                                                                     @Nonnull final EvaluationContext context,
                                                                     @Nullable final byte[] continuation,
                                                                     @Nonnull final ExecuteProperties executeProperties) {
        if (executeProperties.isDryRun()) {
            ...
        } else {
            return RecordCursor.flatMapPipelined(
                    outerContinuation -> getInnerPlan().executePlan(store, context, outerContinuation, executeProperties.clearSkipAndLimit()),
                    (outerQueryResult, innerContinuation) ->
                            RecordCursor.fromFuture(store.deleteRecordAsync(Verify.verifyNotNull(outerQueryResult.getPrimaryKey())))
                                    .filter(isDeleted -> isDeleted)
                                    .map(ignored -> outerQueryResult), // <--------- HERE.
                    continuation, store.getPipelineSize(PipelineOperation.DELETE));
        }
    }
```

So, I refactored its hook in `StoredRecordProperty` to return whatever the inner plan is returning, i.e.:

```java
public Boolean visitDeletePlan(@Nonnull final RecordQueryDeletePlan deletePlan) {
    return storedRecordsFromSingleChild(deletePlan);
}
```